src/pages/assistant/index.tsx

```diff
@@ -47,50 +41,46 @@ export default AssistantPage;
 export const getServerSideProps = withGlobalData(
   withAssistantClient<AssistantLayoutProps & AssistantContentProps>(
     async function ({ client, query }) {
-      const transportModeFilter = parseFilterQuery(query.filter);
-
-      let initialFromFeature;
-      if (query.fromLat && query.fromLon && query.fromLayer) {
-        const fromLat = parseFloat(query.fromLat.toString());
-        const fromLon = parseFloat(query.fromLon.toString());
-        const fromLayer = parseLayerQueryString(query.fromLayer.toString());
-
-        initialFromFeature = await client.reverse(fromLat, fromLon, fromLayer);
-      }
-
-      let initialToFeature;
-      if (query.toLat && query.toLon && query.toLayer) {
-        initialToFeature = await client.reverse(
-          parseFloat(query.toLat.toString()),
-          parseFloat(query.toLon.toString()),
-          parseLayerQueryString(query.toLayer.toString()),
+      const tripQuery = parseTripQuery(query);
+      if (tripQuery) {
+        const from = await client.reverse(
+          parseFloat(tripQuery.fromLat),
+          parseFloat(tripQuery.fromLon),
+          tripQuery.fromLayer,
         );
-      }
-      const arriveBy =
-        query.arriveBy && new Date(Number(query.arriveBy.toString()));
-      const departBy =
-        query.departBy && new Date(Number(query.departBy.toString()));
-      const when = arriveBy || departBy || new Date();
+        const to = await client.reverse(
+          parseFloat(tripQuery.toLat),
+          parseFloat(tripQuery.toLon),
+          tripQuery.toLayer,
+        );
+        const transportModeFilter = parseFilterQuery(tripQuery.filter);
 
-      if (initialFromFeature && initialToFeature) {
-        const trip = await client.trip({
-          from: initialFromFeature,
-          to: initialToFeature,
-          arriveBy: Boolean(arriveBy),
-          when,
-          transportModes: transportModeFilter,
-        });
+        const arriveBy = 'arriveBy' in tripQuery;
+        const departureDate = arriveBy
+          ? new Date(Number(tripQuery.arriveBy))
+          : new Date(Number(tripQuery.departBy)) || new Date();
 
-        return {
-          props: {
-            initialFromFeature,
-            initialToFeature,
-            initialTransportModesFilter: transportModeFilter,
-            trip,
-          },
-        };
-      }
+        if (from && to) {
+          const trip = await client.trip({
+            from,
+            to,
+            ...(arriveBy
+              ? { arriveBy: departureDate }
+              : { departBy: departureDate }),
```

Comment:
Having this multiple places feels like we should do a better abstraction. Like for instance:

```
searchDate: { departureDate, mode: 'arriveBy' }
```

And instead of using boolean all around we have modelled named entities.